### ISSUE TITLE
IncompatibleClassChangeError on Android Lollipop

### ISSUE THREAD

**Issue by inneke-dc** [AlexGilleran/IceSoap#2]
In Crashlytics I see a lot of crashes which all come down to the same thing:

Fatal Exception: java.lang.IncompatibleClassChangeError
Couldn't find com.alexgilleran.icesoap.annotation.XMLField.value
libcore.reflect.AnnotationAccess.toAnnotationInstance (AnnotationAccess.java:659)
java.lang.reflect.Field.getAnnotation (Field.java:242)

For example at lines
com.alexgilleran.icesoap.parser.impl.IceSoapParserImpl.onText (IceSoapParserImpl.java:348)
and 
com.alexgilleran.icesoap.parser.impl.IceSoapParserImpl.hasProcessor (IceSoapParserImpl.java:381)

These crashes all happen on Samsung devices with Android 5. I'm using version 1.0.7 of this library.

Is there any way this can be fixed?

Thanks in advance.

**Comment by AlexGilleran** [comment]
Egh, it's crap like this that made me give up Android development.

Sorry - honestly I've no idea what Samsung are doing to cause this. GSON has the same issue:
https://github.com/google/gson/issues/726

Reported to Samsung here:
http://developer.samsung.com/forum/board/thread/view.do?boardName=General&messageId=280930

Android Issue is here:
https://code.google.com/p/android/issues/detail?id=172339

**Comment by merri-dei** [comment]
@AlexGilleran Did you really give up? I'm starting to feel like giving up, too. We are experiencing the same dilemma with Gson. All happening on samsung devices. It's just so frustrating. Not a single 'solid' work-around is even available.

**Comment by AlexGilleran** [comment]
Yeah, I do web dev in javascript and backend dev in scala now... I got so fed up with Android I ended up buying an iPhone. I have an android app (https://github.com/AlexGilleran/HIITMe) with a 10% crash rate, with all the crash stacktraces deep in Android code with no hope of reproduction. I also find that some percentage of users don't have working sound - once again I can't reproduce it and I have no idea how to fix it, nor any idea of where to start.

Fragmentation in web development is annoying too, but generally a browser incompatibility just results in stuff looking weird - in Android it's nearly always a hard crash as in this case.
